README.md

```diff
@@ -1,2 +1,77 @@
+
+Note the application implementation is deliberately simple (eg System.out instead of a logging framework) to keep it as easy to understand as possible.
+
+## Plugin sub-project
+
+The plugin consists of a [single file](blob/main/plugin/src/main/java/co/elastic/apm/example/webserver/plugin/ExampleHttpServerInstrumentation.java) holding the custom instrumentation, several classes for regression testing, and a pom that builds the correct plugin jar. The details of the plugin project are explained in the articles [creating the instrumentation](https://www.elastic.co/blog/create-your-own-instrumentation-with-the-java-agent-plugin) and [regression testing it](https://www.elastic.co/blog/create-your-own-instrumentation-with-the-java-agent-plugin).
+
+## Building
+
+The full project can be built by cloning to your local system, changing to the root directory, and running `mvn clean install`.
+
+Each sub-project can also be separately built the same way (changing to the sub-project root directory and running `mvn clean install`).
+
+## Running
+
+You need an Elastic APM Java Agent jar (the latest version is recommended, but at least version 1.31.0). Additionally an Elastic APM server is recommended, though not required (communications to the server will be dropped if it's unavailable).
+
+### Standalone Application, no agent, no plugin
+
+You can run the standalone application, *with no agent*, from the project root directory as follows
+
+```aidl
+java -Delastic.apm.service_name=ExampleClient-ExampleAlreadyInstrumentedHttpServer -cp application/target/application-*.jar co.elastic.apm.example.webserver.ExampleClient
+java -Delastic.apm.service_name=ExampleClient-ExampleBasicHttpServer -cp application/target/application-*.jar co.elastic.apm.example.webserver.ExampleClient
+```
+
+### Application with agent, no plugin
+
+Assuming you have set the jar location in `AGENT_JAR` and if available the APM server is specified in `APM_SERVER_URL` and `APM_SECRET_TOKEN`, you can run the application with agent *but no plugin*, from the project root directory, as follows
+
+```aidl
+export AGENT_JAR=...
+export APM_SERVER_URL=...
+export APM_SECRET_TOKEN=...
+
+java -Delastic.apm.server_url=$APM_SERVER_URL -Delastic.apm.secret_token=$APM_SECRET_TOKEN -javaagent:$AGENT_JAR -Delastic.apm.log_level=DEBUG -Delastic.apm.service_name=ExampleClient-ExampleAlreadyInstrumentedHttpServer -cp application/target/application-*.jar co.elastic.apm.example.webserver.ExampleClient
+java -Delastic.apm.server_url=$APM_SERVER_URL -Delastic.apm.secret_token=$APM_SECRET_TOKEN -javaagent:$AGENT_JAR -Delastic.apm.log_level=DEBUG -Delastic.apm.service_name=ExampleClient-ExampleBasicHttpServer -cp application/target/application-*.jar co.elastic.apm.example.webserver.ExampleClient
```

Comment:
[minor] We could probably make this a bit more straightforward for the end user by providing a ready-to-use shell script as it's easy to make a typo or copy-paste errors with long command lines like this.

For users that are on Windows asking them to run it with WSL/bash should not be a big ask as they are developpers.

Reply:
+1